### ISSUE TITLE
Reducing crate_universe lockfile size?

### ISSUE THREAD

**Issue by dae** [bazelbuild/rules_rust#775]
Hi all,

I finally had a chance to experiment properly with crate_universe, and it looks like it'll be a nice improvement over having to commit the BUILD files manually and manually list out dependencies - crates_from() is great!

One thing I noticed is that the lockfile is considerably bigger than Cargo's - in the modest workspace I'm testing with, Cargo's is about 100k, and the universe one is around 1.8MB. When updating dependencies, the changed lines in git appear to be approximately 10x what they are in the Cargo.lock file.

Would it be practical to store the locked versions in a more compact format in the future?

**Comment by dae** [comment]
Sorry, after doing some more digging I found this discussed here: https://github.com/bazelbuild/rules_rust/pull/598#issuecomment-785199849

If using an existing Cargo.lock file were possible, that would be ideal, as that way one would be ensured that the dependencies used in Bazel exactly match those being used via Cargo/Rust Analyzer.

**Comment by illicitonion** [comment]
Yeah, I'm hoping we can get to a point where we're using a `Cargo.lock` file, probably with an additional magic comment containing a hash in it (which I think is fine - if cargo re-generates it and the magic comment is missing, we know we have work to do, but cargo will happily ignore a comment). If you'd be interested in pushing this forwards, I'd be happy to write up some mentoring guidance!

**Comment by dae** [comment]
Bit swamped at the moment, but might take you up on that in the future if no-one else gets to it first. Glad to hear the current lockfile format is more of an implementation detail than a fundamental requirement in any case. :-)
